### DISCUSSION TITLE
Running in a potentially unsafe mode

### DISCUSSION THREAD

**Discussion by mrfelis** [ravendb/ravendb#17207] [General]
We are developing a product using RavenDB.Embedded 5.4. While working on HTTPS support, we passed the application to a couple of inhouse users to test content they are developing to ship with the product. In order to allow HTTP connection, the settings file includes this bit of JSON:

`{"Security.UnsecuredAccessAllowed":"PrivateNetwork", ...`

When we start the embedded server, we pass in the `ServerOptions.ServerUrl = $"http://{Dns.GetHostName()}:{port}";`

Unfortunately, we have a few users that getting the error:

> Running in a potentially unsafe mode. Server certificate information has not been set up and the server address is not configured within allowed unsecured access address range. Please find the RavenDB settings file *settings.json* in the server directory and fill in your certificate information in either Security.Certificate.Path or Security.Certificate.Load.Exec If you would rather like to keep your server unsecured, please relax the Security.UnsecuredAccessAllowed setting to match the ServerUrl setting value.

One user reports that he only sees this error when connected to the company's WIFI. It works for him, when working from home. In trying to reproduce this error, I set up a virtual machine. I can only see this error if I remove the `Security.UnsecuredAccessAllowed` setting and turn on file sharing on. I've messed with all the network settings that I can find and as long as I have this setting set tio "PrivateNetwork", I have no issues.

Does anyone know what could cause RavenDB to fail to start with this error, even when UnsecuredAccessAllowed  is set?

**Comment by ayende** [comment]
The logic here for `PrivateNetwork` is that we'll allow the following IPs:

* 10.x.x.x
* 192.168.x.x
* 172.1[16..31].x.x

Anything else will be considered `PublicNetwork`.

I assume that the company WiFi gives it a different IP address, resulting in this warning.
Note that your setup it _very_ likely that others may be able to reach the server.

Is there a reason that you aren't using `localhost` as the IP, to at least limit the access at the network level?

**Reply by mrfelis** [reply]
Technically, the application is a client/sever app. Some customers will require the client and server run on different machines. In production, customers will be using HTTPS, but we still need to keep the content team working (all internal users, not customers). Security for the content team is not a high priority as they will only be storing test data in RavenDB and not PII or other sensitive information.

**Reply by ayende** [reply]
In that case, make that `PublicNetwork`
